frontend/src/components/DateSelector/DateSelector.js

```diff
@@ -0,0 +1,75 @@
+import { useSelector } from "react-redux";
+import Container from "react-bootstrap/Container";
+import DropdownButton from "react-bootstrap/DropdownButton";
+import Dropdown from "react-bootstrap/Dropdown";
+import { useState } from "react";
+
+function DateSelector() {
+  const [month, setMonth] = useState(
+    new Date().toLocaleString("en-US", { month: "long" })
+  );
+  let maxOffset = 10;
+  let currentYear = new Date().getFullYear();
+  let allYears = [];
+  for (let x = 0; x <= maxOffset; x++) {
```

Comment:
nitpick: if you're going to use single-letters in for-loop use "i" or use a variable